### DISCUSSION TITLE
How to recieve 433 and 868 signals with one Reciever and rtl_433?

### DISCUSSION THREAD

**Discussion by kami83** [merbanan/rtl_433#3069] [Q&A] [answered]
Hi, i would like to decode my Bresser 7in1 Weather sensor (868M) and some different sensor from 433M. Temporaly i am using one rtl-sdr with a raspberry. Can i configure it to recieve the new Bresser sensor?

BR kami

**Comment by HolgerAusB** [comment, marked as answer]
as long, as you have no time critical sensors, aka alarms, you can hop frequencies:

`rtl_433 -f 433.53M -f 686.02M -H 600`

this will hop every 5 minutes (600 seconds) to the other frequency. You might change or strip the decimal part to your needs. And you should test, how often the sensors sending data. Everything above 5 minutes makes it difficult.

And, of course, you will loose all data on the frequency, which is not active in the interval. But for weather stations, that should be OK.

**Reply by kami83** [reply]
Thanks a lot. This works fine.

**Reply by lemek84** [reply]
Can I set a different sample rate for each frequency?

What I've tried:

`rtl_433 -f 433.92M -s 250k -f 868.95M -s 1200k -H 90 -R 91 -R 104 -C si`

but only the last given sampling rate is taken into account.

**Reply by zuckschwerdt** [reply]
There is only one global sample rate. It does not change with hops. If you need more options then you need to use the HTTP-API (`-F http`) to effect hops and e.g. sample rate changes. E.g. `xh :8433/cmd cmd==center_frequency val==433920123` `xh :8433/cmd cmd==sample_rate val==250000` `xh :8433/cmd cmd==gain arg==10` …

**Comment by ghost** [comment]
hi kami83,
-H 600 works, but u loose a lot of messages at 866 Mhz.
try rtl_433 -C si -M level -f 433.92M -f 868.95M -s 1200k -H 25
this works best for Bresser 7in1 Weather sensor and ofc all M-Bus devices
have fun
Boing

**Reply by kami83** [reply]
Hi, thanks a lot for your reply, but i don't understand the difference why do you use -C si, -s 1200k and -M level. I understand that you go down with the hop level, but what about the rest? i just use   -R 42 -R 195 -R 121 -R 173 to get messages from my devices. 

BR kami

**Comment by ghost** [comment]
hi kami,

sry for the overhead.
-C si  converts units f.i. °F to °C  Default is "native" > "Fahrenheit"
-s 1200k is usefull for long telegrams f.i. M-Bus with 150+ Bytes
-M level adds signal level to output

simply use  "rtl_433 rtl_433 -f 433.92M -f 868.95M -H 25"    if u dont need above.

Boing

**Reply by k-str50** [reply]
Hello
I have similar problems with my WS3080 weather station made by ELV.
When I enter the command 
`sudo rtl_433 -R 32 -f 8680000000M `
Is the parameter -R 32 ok? I found two devices 32 and 155. Which ist he correct one?
I only get weather data sporadically. 
What can be the reason for this?
Thank you in advance.
Kurt

**Reply by zuckschwerdt** [reply]
`868M` or `868000k` or `868000000` and perhaps better: `868.3M`

**Reply by k-str50** [reply]
@zuckschwerdt,
Hello
Thank you very much for the answer.
Today it's going very well. It is sunny. 
Could it be that it doesn't work in bad weather? Can I then amplify the signal?
Kurt

**Reply by zuckschwerdt** [reply]
Yes, it's recommended to use: `rtl_433 -Y autolevel -M level -M noise`
"autolevel" will improve reception and "level"/"noise" will show you reception levels.

**Reply by k-str50** [reply]
@zuckschwerdt ,
thank yow very much.
I use this comnand:
`sudo rtl_433 -R 32 -f 868.6M -Y autolevel`
It works fine now. 
Is this ok?
Kurt

**Reply by zuckschwerdt** [reply]
sure, looks good!

**Reply by ghost** [reply]
hi lemek,

if u are on linux:
bash is your friend:

make a script in /home
timehop.sh
#!/bin/bash
#
while :
do
rtl_433 -f 433.92M -s 250k -R 91 -R 104 -C si -T 90
rtl_433 -f 868.95M -s 1200k -R 91 -R 104 -C si -T 90
sleep 1
done

then start ./timehop.sh or better ; let it autostart

if Windows then dunno, but should be possible too

have fun 
Boing



Windows ? dunno, but should be possible..

**Reply by k-str50** [reply]
> sure, looks good!

@zuckschwerdt ,
I'm sorry that I have to ask again.
It seems as if a reception depends on the weather (sunshine). I only get data very sporadically. Could it be that the transmission power depends on the solar radiation? 
Can I possibly adjust the sensitivity?
Kurt

**Reply by zuckschwerdt** [reply]
Use `-M level -M noise` and watch the levels. Noise will tell you if the band is clear (expect around -20 dB, at worst -15 dB) and signal level will show you if autogain works, should be near -0 dB but could be much fainter.

**Reply by k-str50** [reply]
I sent this command : 
`rtl_433 -R 32 -f 868.6M -Y autolevel -M level -M noise`
This is the result: 
`Auto Level] Current noise level -15.2 dB, estimated noise -15.2 dB`
It's not really good.
Should I use autogain?

**Comment by ghost** [comment]
Hi k-str50,
Check your antenna and (or) cable.
Even a very simple dipol for 868 Mhz
gives very good results.
Which RTL device u use?, antenna?
Indoor or Outdoor use?
Check without -R 32 for 5 minutes
rtl_433 -f 868.6M > output.txt
Post output.txt here.
On Linux take Gqrx with 868M
Any Signal?
My Output with a 868 dipol

[102;92m[ [30mSDR [92m] [0m Using device 0: Realtek, RTL2838UHIDIR, SN:
00000001, "Terratec Cinergy T Stick RC (Rev.3)"
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -18.1 dB,
adjusting minimum detection level to -15.1 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -20.8 dB,
adjusting minimum detection level to -17.8 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -23.0 dB,
adjusting minimum detection level to -20.0 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -25.0 dB,
adjusting minimum detection level to -22.0 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -26.7 dB,
adjusting minimum detection level to -23.7 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -28.3 dB,
adjusting minimum detection level to -25.3 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -29.5 dB,
adjusting minimum detection level to -26.5 dB
[103;93m[ [30mAuto Level [93m] [0m Current signal level -14.9 dB, estimated
noise -29.4 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -30.5 dB,
adjusting minimum detection level to -27.5 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -31.6 dB,
adjusting minimum detection level to -28.6 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -32.6 dB,
adjusting minimum detection level to -29.6 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -31.6 dB,
adjusting minimum detection level to -28.6 dB
[103;93m[ [30mAuto Level [93m] [0m Current noise level -32.1 dB, estimated
noise -31.4 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -30.6 dB,
adjusting minimum detection level to -27.6 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -31.7 dB,
adjusting minimum detection level to -28.7 dB
[103;93m[ [30mAuto Level [93m] [0m Estimated noise level is -32.9 dB,
adjusting minimum detection level to -29.9 dB
[103;93m[ [30mAuto Level [93m] [0m Current noise level -32.7 dB, estimated
noise -32.1 dB
Boing

k-str50 ***@***.***> schrieb am Sa., 19. Okt. 2024, 12:30:

**Reply by k-str50** [reply]
@AkaBoing 
Thank you very much for your suggestion.
I have a dongle of the type Realtek 
```

lsusb
Bus 001 Device 004: ID 0bda:2838 Realtek Semiconductor Corp. RTL2838 DVB-T
```
I have attached a picture of the antenna.
![DSCN0019](https://github.com/user-attachments/assets/69c6e6a3-3285-45c8-b7ca-86caac656fea)
The command 
```
rtl_433 -f 868.6M > output.txt
rtl_433 version 23.11-168-g322e04ff branch master at 202410160342 inputs file rtl_tcp RTL-SDR

New defaults active, use "-Y classic -s 250k" if you need the old defaults

Detached kernel driver
Found Fitipower FC0012 tuner
Exact sample rate is: 1000000.026491 Hz
Allocating 15 zero-copy buffers

```
results in this result.
[output.txt](https://github.com/user-attachments/files/17459700/output.txt)

How should I use gqrx? There is a GUI or the cli version. Which one is meant? And what exactly should I do?

Can you please write to me what I should continue to do?
Kurt

**Comment by ghost** [comment]
Hi, k-str50,
Your device RTL 2832U OEM is working.
The antenna is simple but useable.
The output shows 8x Smoke Detektor HAG and one -R 155 Fineoffset - WHx080 ,
all plausible

-R 32 is 433.92M,
take -R 155 instead

On Debian or Ubuntu take the GUI Version of Gqrx, should be in APT repos.
Play with this tool, simple but
very usefull.
have fun
Boing


k-str50 ***@***.***> schrieb am Mo., 21. Okt. 2024, 13:15:

**Comment by ghost** [comment]
moin k-str50
ELV sagt mir: Germany
Die Rauchmelder in deiner Nähe können dein -R 155 Sensorsignal massiv
stören. Also die Antenne möglichst weit weg davon aufstellen.
Gruß
Detlef.

Detlef Haasper ***@***.***> schrieb am Mo., 21. Okt. 2024,
13:56:

**Comment by ghost** [comment]
last attempt;
try  rtl433 -f 868.6M -M level -s 300k >output2.txt
-M level gives Freq1 and Freq2, RSSI,, SNR and NOISE
-s 300k keeps distance from M-Bus at 868.95M
Boing

**Comment by k-str50** [comment]
Hi, I ran this command 
 `sudo rtl_433 -f 868.35M -M level -s 300k -R 32 -R 155 -F kv -F "influx://127.0.0.1:8086/write?db=rtl433" &`
It works fine.
Now I have one more question: 
is it possible to run rtl_433 to write a full record to influxDB every 15 minutes?
Thanks
Kurt

**Reply by zuckschwerdt** [reply]
rtl_433 does not need any privileges, don't ever use `sudo`.

If you need advanced control over the output then best use a relay script, e.g. adapt https://github.com/merbanan/rtl_433/blob/master/examples/rtl_433_influxdb_relay.py

**Comment by k-str50** [comment]
Hello
I would like to thank the people who created this program. 
It's great and works great for me after some help.
Kurt

**Comment by k-str50** [comment]
Hello, everything woks fine. But I would like to store my weather data on the Web. Does any have a Suggestion? Which one is the best? With low cost? 
thanks
Kurt

**Comment by ghost** [comment]
hi k-str50,
like this : ?
Openweather.org
take the One Call API 3.0 .
But this is heavy stuff !
You need a lot of learning time.
If you are a programmer:
 build a Script
rtl_433 JSON to OCA3
or 
a  (simple lol) CGI bridge script and a public http server
will do the job too.
Viele Wege führen nach Rom.
good luck
Boing
good luck

**Comment by k-str50** [comment]
@AkaBoing ,
thanks for your answer. 
I was looking for this. Openweather.org doesn't exist. But I found https://openweathermap.org. Is this ok? What should i use? 
What about influxdb in a cloud?
There is a dirct interface from rtl433 to influx?

**Reply by ghost** [reply]
Sorry for the Missing "map", yes this the correct address. But as i said-  its a lot learning stuff.
And its NOT completly OpenSource AND ofc. Its NOT free. 6 $(¢)per 1000 webcalls atm.
InfluxDB should work (every SQL based solution should work).
Christian can say If a "direct Interface" to InfluxDB is present. I think Not.
Have fun.
Boing

**Reply by zuckschwerdt** [reply]
Yes, InfluxDB is supported:
```
$ rtl_433 -F
...
  [-F log|kv|json|csv|mqtt|influx|syslog|trigger|null] Produce decoded output in given format.
...
	Specify InfluxDB 2.0 server with e.g. -F "influx://localhost:9999/api/v2/write?org=<org>&bucket=<bucket>,token=<authtoken>"
	Specify InfluxDB 1.x server with e.g. -F "influx://localhost:8086/write?db=<db>&p=<password>&u=<user>"
	  Additional parameter -M time:unix:usec:utc for correct timestamps in InfluxDB recommended
```

**Reply by ghost** [reply]
Thx

**Comment by ghost** [comment]
![Log-2024-11-11_10-12-32](https://github.com/user-attachments/assets/863f5cb4-c15b-49bc-9869-1be5b01c15eb)
I made my own interface and a Debian WTK3 app
for all my WM- Bus devices and ofc. temp/humidity Sensors. Just for fun. and learned a lot this way.
Boing

**Comment by k-str50** [comment]
@AkaBoing , @zuckschwerdt ,
thanks for your answer.
I do not want to store my influxDB on a raspberry if possible. Do you have any idea how and where I can store it in a cloud?

**Reply by ghost** [reply]
Influx on Raspi? Bad Idea, this kills your SD Card
In 3 days. And The Performance IS very Bad.

Question: 
How much Sensor Data, which Intervall?
If one Sensor only then simply push the Influx Data to a Cloud of your choice.
Take an 1 hour Intervall for start
I would take a simple endless script loop to feed the Cloud with Data.

A http header in front  , the script will do this, and you can view it with any browser. Ofc you have to publish the adress first.
Little afford, mb. some time for the script.
And never publish foreign M-Bus Data ! This is a Grey Zone. Dont even try !

Have fun
Boing

**Reply by k-str50** [reply]
@AkaBoing ,
thanks for your answer.
I have 22 sensor date 
But they are stored between 5 sekonds and 1 minuites. How can I set the intervall to 15 minutes?

**Comment by ghost** [comment]
A simple example:
a public picture on Google Cloud
https://drive.google.com/file/d/1ADM175ZP_ZBL9RESYCYgFftYMuoOvlYo/view?usp=drivesdk

Could be nearly  any kind of Data but No Windows EXE !
And No executeable Linux App.
Boing

**Comment by ghost** [comment]
The JSON Output of rtl_433  (one of my Oregon Meter)
https://drive.google.com/file/d/1ATPeI_JQf89zJADmbD6ganJtADv_Wx1q/view?usp=drivesdk
Boing

**Comment by k-str50** [comment]
Hello, 
I use a SSD-Drive as storage device  instead a SD-Card. Now I can store  my Data in influx. 
Now I have a problem because I get data ervery few seconds which are stored in the database. 
But this is to much because it bloat my database. 
Is there any way to say to rtl433 to store the data every 15 minutes?
Thanks for a hint.
Kurt

**Reply by zuckschwerdt** [reply]
Not possible with just rtl_433. In Influx you should be able to configure this with things like continuous queries, aggregate functions and data retention policies.

**Reply by ghost** [reply]
try this:
watch -n 900 rtl_433 -f 868.35M -M Level -s 300k -E quit -R155 -F kv -F "influx://127.0.0.1:8086/write?db=rtl433" &

watch with -n 900 starts rtl_433 every 15 mins
rtl_433 with -E quit exits after first good message.
Boing

**Reply by k-str50** [reply]
@AkaBoing @zuckschwerdt,
thanks for your answers. Is there any way to read also a bmp280 device by using rtl433?

**Reply by ghost** [reply]
Bosch BMP280  a nice discussion here:
https://forum.arduino.cc/t/sending-bmp280-data-over-433-mhz/1078118

Boing

**Comment by k-str50** [comment]
Hello,
The program works very well and stably.
I use it in this way:
`rtl_433 -f 868.35M -M level -s 300k -R 32 -R 155 -F kv -F "influx://127.0.0.1:8086/write?db=TestRtl433DB" -H 1800 &`

Unfortunately, I have not yet been able to find a solution for 
1. a setting for an interval e.g. 1800s = 30 minutes 
2. for a readout of a BMP180 sensor.

to find.
Is there something there?

**Reply by zuckschwerdt** [reply]
Hopping here will only work if you also have a config file with e.g. `frequency 433.92M`, otherwise you need `rtl_433 -f 433.92M -f 868.35M -H 1800 …`
Note that `-H 1800` is the same as `-H 1800s` and also `-H 30m` or even `-H 0:30:00` or `-H 0:30`

The BMP180 or BMP280 is a chip not a radio device. It would need to be built into some sender, then that device could be read.

**Reply by kami83** [reply]
Hi, i am still have problems with the Distance of 868 Mhz senders like the Ecowitt WH51. 
I am hoping from 433 to -f 868M and i can only put the senders really near to the external Antenna. Like 2-3m away. Outside. No Walls,etc. I am using a 433mHz Antenna. 
Can someon give me any help to fix this?

BR kami

**Reply by zuckschwerdt** [reply]
Usually `-f 868.3M` works better. Can you try that on it's own? Does range improve then?

**Reply by kami83** [reply]
thanks i will try

**Reply by kami83** [reply]
So after some testing. Same problem. 1-2 meters more and now recieving any messages.

**Reply by zuckschwerdt** [reply]
One option would be to capture transmissions at 1-2 m and also at ~4 m and then compare the spectrogram. You might need to record (`-w rec_868.3M_1000k.cu8`) a longer file and search for the 4 m transmission.

**Reply by kami83** [reply]
mhh okay. I will try it on the weekend. BR kami

**Reply by kami83** [reply]
can you give some more information how to analyse the cu8 file. Or maybe i am using the ecowitt wh51 which are recognized by the rtl_433 decoder, is there any way to analyse the transmission better?  

Thanks a lot.

BR kami

**Reply by zuckschwerdt** [reply]
Try https://triq.org/pdv3/ or https://github.com/triq-org/iqviewer
Compare if the signal is clearly visible and not clipping. Maybe compare to signals from https://github.com/merbanan/rtl_433_tests/

**Reply by kami83** [reply]
Thanks i will try

**Reply by kami83** [reply]
Hi, i am still working on the capture transmission. But the files getting really big. I just want to capture transmission from Fineoffset WH51 (rtl_433 Number 142) for now i am using this cmd: rtl_433 -R 142 -f 868.3M -w rec_868.3M_1000k.cu8 but i cannot find any testing samples for this device. 
Can you also tell me a good Antenna for 868.3 Mhz frequencies. Not to build it on my own just to buy it :)

BR kami

**Reply by zuckschwerdt** [reply]
There are some test samples in https://github.com/merbanan/rtl_433_tests/tree/master/tests/fineoffset/fineoffset_wh51

**Reply by kami83** [reply]
okay but is it problem that they are not for 868 Mhz? Can you help me with my other questions?

**Reply by zuckschwerdt** [reply]
The signals should look the same on any band or is 868 MHz FSK? No idea about antennas, sorry.
